design/ceph-external-cluster.md

```diff
@@ -0,0 +1,105 @@
+1. The admin creates a yaml file with the needed resources from the external cluster (ideally we would provide a helper script to help automate this task):
+   * Save the mon list and admin keyring
+1. Load the yaml file into the local cluster
+   * `oc create -f <config.yaml>`
+
+The CephCluster CRD will have a new property “external” to indicate whether the cluster is external. If true, the local operator will implement the described behavior.
+
+```yaml
+kind: CephCluster
+spec:
+  external: true
+```
+
+The mgr modules, including the dashboard, would be running in the external cluster. Any configuration that happens through the dashboard would depend on the orchestration modules in that external cluster.
+
+## Block Storage
+
+With the rook-ceph cluster created, the CSI driver integration will cover the Block (RWO) storage and no additional management is needed.
+
+## Filesystem (MDS)
+
+A filesystem can be created by Rook that will be specific to the local cluster. This implies that you could have multiple instances of CephFS, with one or more instances of CephFS per K8s cluster that is connecting to the external cluster.
+
+When the CephFilesystem CRD is created in the local cluster, the local Rook operator does the following:
+1. Create the data and metadata pools in the external cluster
+1. Create the filesystem in the external cluster
+1. Start the MDS in the local cluster
+1. Local clients will connect to the local MDS and the external mons and osds
+
+No code changes are necessary for this scenario. The external connections in step 1 and 2 are possible since the local operator knows the mon endpoints and has the admin keyring. For step 3, the MDS started in the local cluster won’t even know that Ceph is running externally. Again, all the MDS needs is the external mon endpoints and the keyring.
+
+One change that might be considered is whether we expect that file systems in different clusters should be independent. If so, a simple prefix could be added to the names of the pools and filesystem to be created in the external cluster. However, we shouldn’t enforce this since it could easily be a desired scenario to share the same instance of CephFS across different clusters.
+
+## Object Storage (RGW)
+
+Similarly to the filesystem, and object store can be created that will be specific to the local cluster. All that is needed is an independent realm name on the external cluster.
+
+When the CephObjectStore CRD is created in the local cluster, the local Rook operator does the following:
+1. Create the metadata and data pools in the external cluster
+1. Create a realm, zone, and zone group in the external cluster
+1. Start the RGW daemon in the local cluster
+1. Local s3 clients will connect to the local RGW and the external mons and osds
```

Comment:
The clients will only connect to the RGW endpoint, the only thing they need is the HTTP to speak S3.

Reply:
Oh right, it is just RGW in this case. I'll update it.

Reply:
No addressed yet.